app/controllers/application_controller.rb

```diff
@@ -13,12 +15,6 @@ def switch_locale(&action)
 
   private
 
-  def adopter_with_profile
-    return if current_user.adopter_account&.adopter_foster_profile
-
-    redirect_to root_path, alert: "Unauthorized action."
-  end
-
   def check_if_adopter
```

Comment:
I think this can be deleted too - looks like it's not referenced anymore

Reply:
Good catch! There were a few other old methods that I could remove as well and just removed in 73a4192a6547a1661bfa1585c5333713eafe3e53.